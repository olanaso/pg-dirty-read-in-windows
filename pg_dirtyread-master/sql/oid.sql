-- test oid columns (removed in PostgreSQL 12)

SELECT setting::int >= 120000 AS is_pg_12 FROM pg_settings WHERE name = 'server_version_num';

SELECT * FROM pg_dirtyread('foo') AS t(oid oid, bar bigint, baz text);

-- error cases
SELECT * FROM pg_dirtyread('foo') as t(oid bigint);
