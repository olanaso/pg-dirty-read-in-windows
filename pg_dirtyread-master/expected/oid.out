-- test oid columns (removed in PostgreSQL 12)
SELECT setting::int >= 120000 AS is_pg_12 FROM pg_settings WHERE name = 'server_version_num';
 is_pg_12 
----------
 t
(1 row)

SELECT * FROM pg_dirtyread('foo') AS t(oid oid, bar bigint, baz text);
ERROR:  Error converting tuple descriptors!
DETAIL:  Attribute "oid" does not exist in type foo.
-- error cases
SELECT * FROM pg_dirtyread('foo') as t(oid bigint);
ERROR:  Error converting tuple descriptors!
DETAIL:  Attribute "oid" does not exist in type foo.
